source/StellarObject.cpp

```diff
@@ -273,7 +289,54 @@ void StellarObject::TryToFire(vector<Projectile> &projectiles, const System *sys
 
 
 
+// Resets ammo after every day.
 void StellarObject::ResetAmmo() const
 {
 	ammo = maxAmmo;
 }
+
+
+
+// Checks if there is a Antimissilelauncher and returns its range.
+int StellarObject::AntiMissile() const
+{
+	if(!launcher)
+		return 0;
+	return launcher->AntiMissile();
+}
+
+
+
+// Tries to fire Antimissilesystem on the projectile.
+bool StellarObject::FireAntiMissile(const Projectile &projectile, vector<Visual> &visuals) const
```

Comment:
It looks like you're duplicating methods (e.g. `Hardpoint::FireAntiMissile`). I would approach this concept (planetary defenses) by using the current classes devoted to weaponry - `Armament` and `Hardpoint`.